File: #04/q01 - fibonnaci/fibonnaci.js

fibonnaci = (x) => x <=1 ? x
            : fibonnaci (x-2) + fibonnaci (x-1)


console.log(fibonnaci(7));
console.log(fibonnaci(0));
console.log(fibonnaci(1));
console.log(fibonnaci(6));
console.log(fibonnaci(5));

File: #04/q01 - fibonnaci/fibonnaci.hs
fibonnaci :: (Eq a, Num a, Num p) => a -> p
fibonnaci 1 = 1
fibonnaci 0 = 0
fibonnaci a = fibonnaci (a - 1) + fibonnaci (a - 2)

main :: IO ()
main = do
    a <- readLn :: IO Int
    print $ fibonnaci a